cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -417,9 +480,9 @@ class self_comparator {
    * @tparam Nullate A cudf::nullate type describing whether to check for nulls.
    */
   template <typename Nullate>
-  device_row_comparator<Nullate> device_comparator(Nullate nullate = {}) const
+  device_less_comparator<Nullate> device_comparator(Nullate nullate = {}) const
```

Comment:
Be sure to check benchmarks. @devavret and I spent a long time puzzling about perf regressions that seemed to stem from adding further layers of wrapping types.

Reply:
[search_new_json.txt](https://github.com/rapidsai/cudf/files/8655336/search_new_json.txt)
[search_old_json.txt](https://github.com/rapidsai/cudf/files/8655337/search_old_json.txt)
[sort_new_json.txt](https://github.com/rapidsai/cudf/files/8655338/sort_new_json.txt)
[sort_old_json.txt](https://github.com/rapidsai/cudf/files/8655339/sort_old_json.txt)

These are the json benchmark files (github won't let me upload json so convert them back from txt). There's only 1 concerning benchmark that Bradley and I were attributing to noise due to its small size:
`Search/Table/2/1000/manual_time                   +0.0524     +0.0379       0       0       0       0`

Reply:
Reran benchmark with @jrhemstad fold command, nearly identical results with the recursive parameter unpack:
[search_new_fold_json.txt](https://github.com/rapidsai/cudf/files/8655468/search_new_fold_json.txt)
[sort_new_fold_json.txt](https://github.com/rapidsai/cudf/files/8655469/sort_new_fold_json.txt)
`Search/Table/2/1000/manual_time                                    +0.0556         +0.0406             0             0             0             0`